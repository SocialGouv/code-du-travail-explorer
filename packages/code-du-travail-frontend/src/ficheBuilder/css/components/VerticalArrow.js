import styled from "styled-components";

const VerticalArrow = styled.div.attrs(() => ({
  role: "presentation"
}))`
  display: inline-block;
  position: relative;
  width: 24px;
  height: 12px;
  position: absolute;
  top: 50%;
  right: 0;
  margin-top: -6px;

  &::after,
  &::before {
    display: block;
    position: absolute;
    top: 50%;
    width: 10px;
    height: 2px;
    background-color: currentColor;
    content: "";
  }

  &::before {
    left: 4px;
    transform: rotate(45deg);
  }

  [aria-expanded="true"] &::before,
  [aria-selected="true"] &::before {
    transform: rotate(-45deg);
  }

  &::after {
    right: 4px;
    transform: rotate(-45deg);
  }

  [aria-expanded="true"] &::after,
  [aria-selected="true"] &::after {
    transform: rotate(45deg);
  }

  &::before,
  &::after {
    transition: transform 0.25s ease, -webkit-transform 0.25s ease;
  }
`;

export default VerticalArrow;
