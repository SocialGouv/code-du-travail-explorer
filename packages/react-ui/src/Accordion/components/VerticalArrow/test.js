import { render } from "@testing-library/react";
import React from "react";

import { VerticalArrow } from ".";

describe("<VerticalArrow />", () => {
  test("should render", () => {
    const { container } = render(<VerticalArrow />);
    expect(container).toMatchSnapshot();
  });
});
