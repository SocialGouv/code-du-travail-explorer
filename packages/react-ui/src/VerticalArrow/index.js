import styled from "styled-components";

export const VerticalArrow = styled.div.attrs(() => ({
  role: "presentation"
}))`
  position: relative;
  display: inline-block;
  flex: 0 0 auto;
  width: 24px;
  height: 12px;

  &::after,
  &::before {
    position: absolute;
    top: 50%;
    display: block;
    width: 10px;
    height: 2px;
    background-color: currentColor;
    transition: transform 0.25s ease, -webkit-transform 0.25s ease;
    content: "";
  }

  &::before {
    left: 4px;
    transform: rotate(45deg);
  }

  [aria-expanded="true"] &::before,
  [aria-selected="true"] &::before {
    transform: rotate(-45deg);
  }

  &::after {
    right: 4px;
    transform: rotate(-45deg);
  }

  [aria-expanded="true"] &::after,
  [aria-selected="true"] &::after {
    transform: rotate(45deg);
  }
`;
