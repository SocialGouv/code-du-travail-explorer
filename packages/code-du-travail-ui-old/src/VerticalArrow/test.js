import React from "react";
import { render } from "@testing-library/react";
import VerticalArrow from ".";

describe("<VerticalArrow />", () => {
  test("should render", () => {
    const { container } = render(<VerticalArrow />);
    expect(container).toMatchSnapshot();
  });
});
