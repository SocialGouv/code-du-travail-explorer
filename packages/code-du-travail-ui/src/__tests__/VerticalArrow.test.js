import React from "react";
import { render } from "react-testing-library";
import VerticalArrow from "../VerticalArrow";

describe("<VerticalArrow />", () => {
  test("should render", () => {
    const { container } = render(<VerticalArrow />);
    expect(container).toMatchSnapshot();
  });
});
