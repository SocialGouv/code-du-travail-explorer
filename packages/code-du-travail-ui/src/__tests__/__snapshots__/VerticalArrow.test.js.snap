// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerticalArrow /> should render 1`] = `
<div>
  <div
    class="sc-bdVaJa kVneBu"
    role="presentation"
  />
</div>
`;
